### PR TITLE
GH-1895 and GH-1896: Midnight promos

### PR DESCRIPTION
**Pull request by wlycdgr**
Implements the hub and panel Premium promo modals, which replace the initial and upgrade Plus promo modals.

Does not include the GH-1899 telemetry ticket, which will get its own PR.

* [X] Have you followed the guidelines in [CONTRIBUTING.md](../CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you added an explanation of what your changes do?
* [X] Does your submission pass tests?
* [X] Did you lint your code prior to submission?

### CODE REVIEW

File: app/panel/components/Panel.jsx [hunk 1]
```diff
@@ -336,28 +338,36 @@ class Panel extends React.Component {
...
+	 * Renders either the Insights or the Premium promo modal
+	 */
 	_renderPromoModal = () => {
 		const {
 			promoModal,
 			isPromoModalHidden,
 		} = this.props;
 
+		console.error(`promoModal: ${promoModal}`);
+		console.error(`isPromoModalHidden: ${isPromoModalHidden}`);
```
Comment (jsignanini) [inline app/panel/components/Panel.jsx]: @wlycdgr please remove these

File: manifest.json [hunk 2]
```diff
@@ -6,6 +6,8 @@
 			"strict_min_version": "52.0"
 		}
 	},
+  	"debug": true,
+  	"log": true,
```
Comment (jsignanini) [inline manifest.json]: @wlycdgr let's remove these as well please

File: app/panel/components/Panel.jsx (hunk 1 above)
Comment (wlycdgr) [reply app/panel/components/Panel.jsx]: Done

File: manifest.json (hunk 2 above)
Comment (wlycdgr) [reply manifest.json]: Done

**Comment by wlycdgr** [comment]
Premium subscribers should no longer see promo in either hub or panel, if my smoke tests aren't lying

**Pull request merged**